# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/events/test/pointerevents/test_getCoalescedEvents.html
# Commit: 0ea945553f63
# Full Hash: 0ea945553f63659b99fd37265c5e7a764f70504d
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-12-03 16:21:08
# Regressor Bug: 1680669
# File Overlap Count: 1
# Description:
#   Bug 1680669 - part 6: Make methods of `MouseEvent` and static methods of `Event` aware of fractional coordinates r=smaug,emilio,webdriver-reviewers
#   
#   `screenX`, `screenY`, `clientX`, `clientY`, `x`, `y`, `offsetX` and `offsetY`
#   are now `double`.  This patch makes the event classes aware of DOM events.
#   
# ==============================================================================

diff -r 98d5a05c6c6e -r 0ea945553f63 dom/events/test/pointerevents/test_getCoalescedEvents.html
--- a/dom/events/test/pointerevents/test_getCoalescedEvents.html	Tue Dec 03 07:10:01 2024 +0000
+++ b/dom/events/test/pointerevents/test_getCoalescedEvents.html	Tue Dec 03 07:10:05 2024 +0000
@@ -32,8 +32,24 @@
       ok(length >= 1, "Coalesced events should >= 1, got " + length);
 
       let rect = target0.getBoundingClientRect();
-      let prevOffsetX = 0;
-      let prevOffsetY = 0;
+      let prevOffsetX = undefined;
+      let prevOffsetY = undefined;
+
+      function isExpectedOffset(aNewOffset, aPrevOffset) {
+        if (aPrevOffset === undefined) {
+          const roundedOffset = 5 * Math.max(Math.round(aNewOffset / 5), 1);
+          return aNewOffset >= roundedOffset - 1 && aNewOffset <= roundedOffset + 1;
+        }
+        let candidateOffset = aPrevOffset + 5;
+        while (candidateOffset < 25) {
+          // Allow rounding issue.
+          if (aNewOffset >= candidateOffset - 0.1 && aNewOffset <= candidateOffset + 0.1) {
+            return true;
+          }
+          candidateOffset += 5;
+        }
+        return false;
+      }
 
       for (let i = 0; i < length; ++i) {
         let coalescedEvent = ev.getCoalescedEvents()[i];
@@ -48,13 +64,18 @@
         is(coalescedEvent.cancelable, false, "getCoalescedEvents()[" + i + "].cancelable");
         is(coalescedEvent.bubbles, false, "getCoalescedEvents()[" + i + "].bubbles");
 
-        ok(coalescedEvent.offsetX >= prevOffsetX, "getCoalescedEvents()[" + i + "].offsetX = " + coalescedEvent.offsetX);
-        ok(coalescedEvent.offsetX == 5 || coalescedEvent.offsetX == 10 ||
-            coalescedEvent.offsetX == 15 || coalescedEvent.offsetX == 20, "expected offsetX");
-
-        ok(coalescedEvent.offsetY >= prevOffsetY, "getCoalescedEvents()[" + i + "].offsetY = " + coalescedEvent.offsetY);
-        ok(coalescedEvent.offsetY == 5 || coalescedEvent.offsetY == 10 ||
-            coalescedEvent.offsetY == 15 || coalescedEvent.offsetY == 20, "expected offsetY");
+        ok(
+          isExpectedOffset(coalescedEvent.offsetX, prevOffsetX),
+          `getCoalescedEvents()[${i}].offsetX (${
+            coalescedEvent.offsetX
+          }) should be 5 * n + previous offsetX (${prevOffsetX})`
+        );
+        ok(
+          isExpectedOffset(coalescedEvent.offsetY, prevOffsetY),
+          `getCoalescedEvents()[${i}].offsetY (${
+            coalescedEvent.offsetY
+          }) should be 5 * n + previous offsetY (${prevOffsetY})`
+        );
 
         prevOffsetX = coalescedEvent.offsetX;
         prevOffsetY = coalescedEvent.offsetY;